### ISSUE TITLE
Reactive WebClient: new Span with new spanId seems to be generated inside http request

### ISSUE THREAD

**Issue by bjuchems** [micrometer-metrics/tracing#316]
I am currently migrating from Spring 2.7 to 3.x, so I need to migrate from Sleuth to Micrometer.

We have following constellation:
Service A calls Service B per Reactive WebClient (Netty).

In Sleuth we had had the following option:

Service A does several parallel RPC requests and has generated the root span (traceId: 1, spanId: 2)
We log traceId, spanId and additionally parentId (I explain why we do that later): [traceId, parentId, spanId]

Service A has a Span with traceId and spanId.
Service A logs before request: [1, ,2]
Service A calls Service B. 
Service B extracts B3-Header creates a new spanId and sets given spanId as parentId.
Service B logs; [1, 2, 3]
Service A logs after request: [1, ,2]

So, spanId 2 may be traced between service.

That has worked with Sleuth.
With Micrometer it does not work anymore. Current situation:
Service A logs: [1, ,2]
Service A calls Service B.
Service A somewhere deep inside reactive WebClient generates sets a new spanId: 4
Service B extracts B3-Header creates a new spanId and sets given spanId as parentId.
Service B logs: [1, 4, 3]
Service A logs after request: [1, ,2]

SpanId may not be traced between services anymore. I do not find any possibility to get a connection between spanId 4 and spanId 2.

For demonstration I added some logging to WebClient:
```
log.info("before webClient.get()");
		return webClient.get()
				.uri("myUri")
				.retrieve()
				.bodyToMono(MyClass.class)
				.doOnEach(signal -> {
					log.info("webClient.doOnEach() - signal: {}", signal.getType());
				})
				.doOnRequest(longValue -> {
					log.info("webClient.doOnRequest() - long value: {}", longValue);
				})
				.block();
(...)
log.info("before webClient.get()");
```
and to the controller function:
```
log.info("extracted b3 header: {}", httpServletRequest.getHeader("b3"));
```

These are the resulting logs:

```
2023-07-26 14:12:00.006  INFO [64c10d8fbf6496ec947100001fef1b94, ,a1b960ab4ec2795c] 40368 --- [nio-7005-exec-1] serviceA_webClient    : before webClient.get()
2023-07-26 14:12:00.011  INFO [64c10d8fbf6496ec947100001fef1b94, ,a1b960ab4ec2795c] 40368 --- [nio-7005-exec-1] serviceA_webClient    : webClient.doOnRequest() - long value: 9223372036854775807
2023-07-26 14:12:00.340  INFO [64c10d8fbf6496ec947100001fef1b94, ,a1b960ab4ec2795c] 40368 --- [ctor-http-nio-3] serviceA_webClient    : webClient.doOnEach() - signal: onNext
2023-07-26 14:12:00.340  INFO [64c10d8fbf6496ec947100001fef1b94, ,a1b960ab4ec2795c] 40368 --- [ctor-http-nio-3] serviceA_webClient    : webClient.doOnEach() - signal: onComplete
2023-07-26 14:12:00.341  INFO [64c10d8fbf6496ec947100001fef1b94, ,a1b960ab4ec2795c] 40368 --- [nio-7005-exec-1] serviceA_webClient  : after webClient.get()


2023-07-26 14:12:00.061  INFO [64c10d8fbf6496ec947100001fef1b94,59057f7a94de3c4b,5dc44d1baf0a8d3d] 13572 --- [nio-7000-exec-7] serviceB_controller    : extracted b3 header: 64c10d8fbf6496ec947100001fef1b94-59057f7a94de3c4b-1
```

We can see: traceId is propagated correctly, but the spanId "2" from Service A ("a1b960ab4ec2795c") is not found on Service B side, instead of we have a new spanId "4" ("59057f7a94de3c4b").

Where does this new spanId come from?
Is there a configuration option to get back to the old way?

**Comment by jonatan-ivanov** [comment]
I don't think this issue has anything to do with Micrometer Tracing but with Boot and WebClient. In Boot 3 the default propagation format was changed to W3C, if you want to use B3, you need to set the right output format on your apps, see: https://github.com/spring-projects/spring-boot/pull/35611

**Comment by bjuchems** [comment]
But, this is not my problem:

I HAVE set logging format to B3 on sending service:
```
management.tracing.propagation.type=B3
```
The receiving service DOES get the correct format, as I stated in my original post:

```
extracted b3 header: 64c10d8fbf6496ec947100001fef1b94-59057f7a94de3c4b-1
```

The problem is not the B3 format. The problem is, that I do have an unknown span id.

**Comment by jonatan-ivanov** [comment]
Oh, sorry, for some reason I thought you have a broken trace because propagation does not work (I also find the above a little hard to follow).

So here's what should happen:
- `Service A` starts processing, creates initial span: traceId=0, spanId=1, parent=null
- `Service A` sends out a request to `Service B`, this will result in a new span, since WebClient is instrumented: traceId=0, spanId=2, parent=1
- `Service B` receives the request, this will result in a new span, since MVC/WebFlux is instrumented: traceId=0, spanId=3, parent=2
(`reactor-netty` is also instrumented so you might have more spans)

Could you please send us a screenshot about what you see in Zipkin (or any other tracing UI) and tell us what should be different?

If your issue is the extra span, WebClient instrumentation can be disabled, please check the docs of Spring Framework (same for reactive-netty).

If your issue is a disconnected trace, could you please create an issue in Spring Framework with a minimal java reproducer?

**Comment by shakuzen** [comment]
@bjuchems are you saying you want the client and server to share a span id, as described in Brave documentation [here](https://github.com/openzipkin/brave/tree/master/brave#sharing-span-ids-between-client-and-server)? If so, I believe you can set this Spring Boot configuration property: [`management.tracing.brave.span-joining-supported`](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#application-properties.actuator.management.tracing.brave.span-joining-supported)
